units/github_status_refresh.go

```diff
@@ -0,0 +1,259 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/send"
+	"github.com/mongodb/grip/sometimes"
+	"github.com/pkg/errors"
+)
+
+const (
+	githubStatusRefreshJobName = "github-status-refresh"
+)
+
+func init() {
+	registry.AddJobType(githubStatusUpdateJobName, func() amboy.Job { return makeGithubStatusRefreshJob() })
+}
+
+// NewGithubStatusRefreshJob is a job that re-sends github statuses to the PR associated with the given patch.
+func NewGithubStatusRefreshJob(p *patch.Patch) amboy.Job {
+	job := makeGithubStatusRefreshJob()
+	job.FetchID = p.Version
+	job.patch = p
+
+	job.SetID(fmt.Sprintf("%s:%s-%s", githubStatusRefreshJobName, p.Version, time.Now().String()))
+	return job
+}
+
+type githubStatusRefreshJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	env      evergreen.Environment
+	sender   send.Sender
+
+	urlBase      string
+	patch        *patch.Patch
+	builds       []build.Build
+	childPatches []patch.Patch
+
+	FetchID string `bson:"fetch_id" json:"fetch_id" yaml:"fetch_id"`
+}
+
+func makeGithubStatusRefreshJob() *githubStatusRefreshJob {
+	j := &githubStatusRefreshJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    githubStatusUpdateJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetPriority(1)
+	return j
+}
+
+func (j *githubStatusRefreshJob) shouldUpdate() (bool, error) {
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		return false, errors.Wrap(err, "error retrieving admin settings")
+	}
+	if flags.GithubStatusAPIDisabled {
+		grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+			"job":     j.Name,
+			"message": "GitHub status updates are disabled, not updating status",
+		})
+		return false, nil
+	}
+	return true, nil
+}
+
+func (j *githubStatusRefreshJob) fetch() error {
```

Comment:
seems that a more fitting name would be something like `populate` since we're populating the job struct's data here